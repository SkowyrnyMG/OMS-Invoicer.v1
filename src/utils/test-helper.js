import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
