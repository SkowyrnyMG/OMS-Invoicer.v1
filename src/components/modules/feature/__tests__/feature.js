import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import Feature from '../feature';

describe('Feature', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(
      <Feature title='Test feature' counter={1}>
        <div />
      </Feature>,
    );
  });
});
