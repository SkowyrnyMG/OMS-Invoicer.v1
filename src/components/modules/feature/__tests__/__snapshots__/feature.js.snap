// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature should display in the document with correct styles 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5rem;
  background-color: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 0 10px -4px rgba(0,0,0,0.25);
  overflow: hidden;
}

.c1 {
  max-height: 10rem;
  margin-bottom: 3rem;
}

.c1 svg {
  width: 8rem !important;
  fill: #446DF6;
}

.c2 {
  font-size: 18px;
}

.c3 {
  position: absolute;
  bottom: -4%;
  line-height: 0.8;
  left: -2%;
  font-family: 'Montserrat';
  color: #446DF644;
  font-weight: 700;
  font-size: 180px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div />
  </div>
  <h3
    class="c2"
  >
    Test feature
  </h3>
  <div
    class="c3"
  >
    1
  </div>
</div>
`;
